crates/router/src/connector/shift4/transformers.rs

```diff
@@ -29,42 +75,121 @@ pub struct Card {
 }
 
 impl TryFrom<&types::PaymentsAuthorizeRouterData> for Shift4PaymentsRequest {
-    type Error = error_stack::Report<errors::ConnectorError>;
+    type Error = Error;
     fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
-        match item.request.payment_method_data.clone() {
-            api::PaymentMethodData::Card(ccard) => {
-                let submit_for_settlement = matches!(
-                    item.request.capture_method,
-                    Some(enums::CaptureMethod::Automatic) | None
-                );
-                let payment_request = Self {
-                    amount: item.request.amount.to_string(),
-                    card: Card {
-                        number: ccard.card_number,
-                        exp_month: ccard.card_exp_month,
-                        exp_year: ccard.card_exp_year,
-                        cardholder_name: ccard.card_holder_name,
-                    },
-                    currency: item.request.currency.to_string(),
-                    description: item.description.clone(),
-                    captured: submit_for_settlement,
-                };
-                Ok(payment_request)
+        match &item.request.payment_method_data {
+            api::PaymentMethodData::Card(ccard) => get_card_payment_request(item, ccard),
+            api::PaymentMethodData::BankRedirect(redirect_data) => {
+                get_bank_redirect_request(item, redirect_data)
             }
-            _ => Err(
-                errors::ConnectorError::NotImplemented("Current Payment Method".to_string()).into(),
-            ),
+            _ => Err(errors::ConnectorError::NotImplemented("Payment Method".to_string()).into()),
```

Comment:
In these two functions, only payment method data are changed and rest of the fields remain the same, can this duplication be avoided by making this function return only the payment method related data and then construct the Shift4 request outside?